godtools/App/Services/Renderer/Views/MobileContent/Views/ContentButton/MobileContentButtonViewModel.swift

```diff
@@ -29,34 +29,44 @@ class MobileContentButtonViewModel: NSObject, MobileContentButtonViewModelType {
+    required init(buttonModel: Button, renderedPageContext: MobileContentRenderedPageContext, mobileContentAnalytics: MobileContentAnalytics, fontService: FontService) {
         
         self.buttonModel = buttonModel
-        self.rendererPageModel = rendererPageModel
-        self.containerModel = containerModel
+        self.renderedPageContext = renderedPageContext
         self.mobileContentAnalytics = mobileContentAnalytics
         self.fontService = fontService
         
-        buttonWidth = MobileContentViewWidth(dimension: buttonModel.buttonWidth)
+        buttonWidth = MobileContentViewWidth(dimension: buttonModel.width)
         
-        let buttonColor: UIColor = buttonModel.getColor()?.uiColor ?? containerModel?.buttonColor?.uiColor ?? rendererPageModel.pageColors.primaryColor.uiColor
-        let buttonTitleColor: UIColor? = buttonModel.getTextColor()?.uiColor
-        
-        let buttonStyle: MobileContentButtonStyle = buttonModel.style ?? containerModel?.buttonStyle ?? .contained
-        
-        switch buttonStyle {
+        let defaultBackgroundColor: UIColor = buttonModel.buttonColor
+        let defaultTitleColor: UIColor = buttonModel.text?.textColor ?? renderedPageContext.pageModel.primaryTextColor
+        let defaultBorderColor: UIColor = .clear
+                                
+        switch buttonModel.style {
         
         case .contained:
-            backgroundColor = buttonColor
-            titleColor = buttonTitleColor ?? rendererPageModel.pageColors.primaryTextColor.uiColor
-            borderColor = UIColor.clear
+            backgroundColor = defaultBackgroundColor
+            titleColor = defaultTitleColor
+            borderColor = defaultBorderColor
+            
         case .outlined:
-            backgroundColor = buttonModel.getBackgroundColor()?.uiColor ?? .clear
-            titleColor = buttonColor
-            borderColor = buttonColor
+            backgroundColor = buttonModel.backgroundColor
+            titleColor = buttonModel.buttonColor
+            borderColor = buttonModel.buttonColor
+            
+        case .unknown:
+            backgroundColor = defaultBackgroundColor
+            titleColor = defaultTitleColor
+            borderColor = defaultBorderColor
+            
+        default:
+            backgroundColor = defaultBackgroundColor
+            titleColor = defaultTitleColor
+            borderColor = defaultBorderColor
         }
-        
-        if let name = buttonModel.iconName, let image = rendererPageModel.resourcesCache.getImageFromManifestResources(fileName: name)  {
+                
+        if let name = buttonModel.icon?.name,
+            let image = renderedPageContext.resourcesCache.getImageFromManifestResources(fileName: name)  {
```

Comment:
an optimization you could make is make `getImageFromManifestResources()` take the `Resource` object directly, that way you don't have to re-lookup the resource object inside that logic.

Reply:
@frett I like this idea.

Reply:
Hey @frett just committed this change.